File: apis/src/main/java/com/beat/apis/performance/application/dto/create/PerformanceRequest.java
package com.beat.apis.performance.application.dto.create;

import java.util.List;

import com.beat.apis.performance.application.dto.BankNameType;
import com.beat.apis.performance.application.dto.GenreType;

import jakarta.validation.constraints.Size;
public record PerformanceRequest(
	String performanceTitle,
	GenreType genre,
	int runningTime,
	@Size(max = 1500, message = "공연 소개는 1500자를 초과할 수 없습니다.")
	String performanceDescription,
	@Size(max = 1500, message = "공연 유의사항은 1500자를 초과할 수 없습니다.")
	String performanceAttentionNote,
	BankNameType bankName,
	String accountNumber,
	String accountHolder,
	String posterImage,
	String performanceTeamName,
	String performanceVenue,
	String roadAddressName,
	String placeDetailAddress,
	String latitude,
	String longitude,
	String performanceContact,
	String performancePeriod,
	int ticketPrice,
	int totalScheduleCount,
	List<ScheduleRequest> scheduleList,
	List<CastRequest> castList,
	List<StaffRequest> staffList,
	List<PerformanceImageRequest> performanceImageList
) {
}

File: apis/src/main/kotlin/com/beat/apis/performance/api/request/PerformanceRequest.kt
package com.beat.apis.performance.api.request

import com.beat.apis.performance.api.type.BankNameType
import com.beat.apis.performance.api.type.GenreType
import jakarta.validation.Valid
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size

data class PerformanceRequest(
    @field:NotNull val performanceTitle: String?,
    @field:NotNull val genre: GenreType?,
    @field:NotNull val runningTime: Int?,
    @field:NotNull @field:Size(max = 1500, message = "공연 소개는 1500자를 초과할 수 없습니다.")
    val performanceDescription: String?,
    @field:NotNull @field:Size(max = 1500, message = "공연 유의사항은 1500자를 초과할 수 없습니다.")
    val performanceAttentionNote: String?,
    val bankName: BankNameType?,
    val accountNumber: String?,
    val accountHolder: String?,
    @field:NotNull val posterImage: String?,
    @field:NotNull val performanceTeamName: String?,
    @field:NotNull val performanceVenue: String?,
    @field:NotNull val roadAddressName: String?,
    @field:NotNull val placeDetailAddress: String?,
    @field:NotNull val latitude: String?,
    @field:NotNull val longitude: String?,
    @field:NotNull val performanceContact: String?,
    @field:NotNull val performancePeriod: String?,
    @field:NotNull val ticketPrice: Int?,
    @field:NotNull val totalScheduleCount: Int?,
    @field:NotNull @field:Valid val scheduleList: List<@Valid ScheduleRequest>?,
    @field:NotNull @field:Valid val castList: List<@Valid CastRequest>?,
    @field:NotNull @field:Valid val staffList: List<@Valid StaffRequest>?,
    @field:NotNull @field:Valid val performanceImageList: List<@Valid PerformanceImageRequest>?,
)
